pkg/handlers/v1/filter.go

```diff
@@ -36,64 +34,80 @@ type AssessmentResult struct {
 	Protocol string `json:"protocol"`
 }
 
-// NexposeVulnFilter accepts a payload with Nexpose asset information
+// FilterHandler accepts a payload with Nexpose asset information
 // and a list of vulnerabilities and returns a payload of the same shape
 // omitting vulnerabilities that do not meet the filter criteria
-type NexposeVulnFilter struct {
-	VulnerabilityFilterCriteria *filter.VulnerabilityFilterCriteria
-	LogFn                       domain.LogFn
-	StatFn                      domain.StatFn
+type FilterHandler struct {
+	VulnerabilityFilter domain.VulnerabilityFilter
+	Producer            domain.Producer
+	LogFn               domain.LogFn
+	StatFn              domain.StatFn
 }
 
-// Handle filters an asset's vulnerabilities based on predefined criteria and returns
-// the payload without vulnerabilities that meet the requirements
-func (h NexposeVulnFilter) Handle(ctx context.Context, input NexposeAssetVulnerabilitiesEvent) (NexposeAssetVulnerabilitiesEvent, error) {
-	output := NexposeAssetVulnerabilitiesEvent{
+// Handle filters any AssetVulnerabilityDetails items from a given NexposeAssetVulnerabilitiesEvent
+// that do not meet the filter criteria, produces the filtered AssetVulnerabilityDetailsEvent to a stream,
+// and returns the filtered AssetVulnerabilityDetailsEvent, or an error if one occurred.
+func (h FilterHandler) Handle(ctx context.Context, input NexposeAssetVulnerabilitiesEvent) (NexposeAssetVulnerabilitiesEvent, error) {
+	asset := domain.Asset{
+		ID:          input.ID,
+		IP:          input.IP,
+		Hostname:    input.Hostname,
+		LastScanned: input.LastScanned,
+	}
+	vulns := vulnDetailsToVuln(input.Vulnerabilities)
+	vulns = h.VulnerabilityFilter.FilterVulnerabilities(ctx, asset, vulns)
+	vulnDetails := vulnToVulnDetails(vulns)
+
+	filteredAssetVulnEvent := NexposeAssetVulnerabilitiesEvent{
 		LastScanned:     input.LastScanned,
 		Hostname:        input.Hostname,
 		ID:              input.ID,
 		IP:              input.IP,
-		Vulnerabilities: h.FilterVulnerabilities(ctx, input),
+		Vulnerabilities: vulnDetails,
+	}
+	_, err := h.Producer.Produce(ctx, filteredAssetVulnEvent)
+	if err != nil {
+		return NexposeAssetVulnerabilitiesEvent{}, err
 	}
-	return output, nil
+	return filteredAssetVulnEvent, nil
 }
 
-// FilterVulnerabilities returns a filtered list of vulnerabilities.
-func (h NexposeVulnFilter) FilterVulnerabilities(ctx context.Context, assetVulnerabilities NexposeAssetVulnerabilitiesEvent) []AssetVulnerabilityDetails {
-	logger := h.LogFn(ctx)
-	stater := h.StatFn(ctx)
-
-	minCvssV2Score := h.VulnerabilityFilterCriteria.CVSSV2MinimumScore
-	vulnIDRegexp := h.VulnerabilityFilterCriteria.VulnIDRegexp
+func vulnDetailsToVuln(vulnDetails []AssetVulnerabilityDetails) []domain.Vulnerability {
+	vulns := make([]domain.Vulnerability, len(vulnDetails))
+	for i, vulnDetail := range vulnDetails {
```

Comment:
nit: depending on the font/typeface, I sometimes have legitimate trouble differentiating between the `i` `l` and `1` characters. Not a big deal, just a nudge for future commits.